plutus-core/untyped-plutus-core/Language/UntypedPlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -574,6 +575,11 @@ unsafeRunCekNoEmit
     -> (EvaluationResult (Term Name uni fun ()), CekExBudgetState fun)
 unsafeRunCekNoEmit runtime mode = first unsafeExtractEvaluationResult . runCekNoEmit runtime mode
 
+-- | When we want to just evaluate the program we use the 'Restricting' mode with an enormous
+-- budget, so that evaluation costs of on-chain budgeting are reflected accurately in benchmarks.
+enormousBudget :: ExBudgetMode
+enormousBudget = Restricting . ExRestrictingBudget $ ExBudget (10^(10::Int)) (10^(10::Int))
+
```

Comment:
It might be handy to give plc an option (`-R` or something) to run a program with an enormous budget.  

It's also maybe not that enormous: testing primality of a 100-digit prime uses 1764437096 ExCPU and 4452490102  ExMemory (both >10^9 but < 10^10), which is getting close.  I think the numbers for validations are all 5-6 digits though,  so that's not really a problem.  Also, using even bigger numbers adds extra costs anyway.

It's possible that I might use costs on a bigger scale at some point (I have a vague idea that that might enable us to avoid floating point computations in the cost functions and possibly save some time), so we might have to make these bigger later (mostly writing this as a reminder to myself).

Reply:
> It might be handy to give plc an option (-R or something) to run a program with an enormous budget.

Done.